### ISSUE TITLE
head-tail overlaps not collapsed correctly

### ISSUE THREAD

**Issue by lsterck** [bcgsc/abyss#268]
When I scaffold my contigs in the abyss pipeline I see a number of warnings being outputted about head and tail contig mismatching in the scaffolding stage.
a few examples:
```
warning: the head of 81131854+ does not match the tail of the previous contig
gagttcactaagatattatttatgtgttactttacttgaagaGGTTTGGTATTTAATAAAGTGAATATATATATATATATATAT
TTCACTAAGATATTATTTATGTGTTACTTACTAAGATATTATTTATGTGTTACTTTACTCGAAGAGGTTTGGTATTTAATAAAG
warning: the head of 154467949- does not match the tail of the previous contig
CATAGATTTACAAAAATCCaacaactataa
AACAACTATAACATAGATTTACAAAAATCC
```
as you can observe the ends are actually (nearly) identical, with a certain shift. 
this issue is a  'branch' from https://github.com/bcgsc/abyss/issues/258 , where it was raised that it might be the lowercase causing the issue. In the cases I encountered it indeed looked like this accounts for many of them. An additional remark is that I think this also is caused by the use of IUPAC code bases, as shown below:
```
warning: the head of 177027237+ does not match the tail of the previous contig
AAATTAGCTTGAGACCTGGTGGAGGAAGCAAATCGGGCTCTAACGAGAACCAAATTGGCTCGAGACCTGGTGGAGGAAGCAAAT
aacgagaaccAAATTRGCTYGAGACCTGGTGGAGGAAGCAAATCGGGCTCTAACGAGAAYCAAATTRGCTCGAGACCTGGTGGA
```
not sure if it's relevant but I must add that the data I'm using for scaffolding (== the MP libraries) are not super clean data (and even contain a fraction of 454 reads).

I had the issue when running abyss v2.1.3 (I started a run to see if it also occurs in other versions)

**Comment by sjackman** [comment]
The very similar sequences are a known deficiency in ABySS, but not a bug exactly. It's an issue that I would like to address.
The identical sequences are a bug. Perhaps the test is not case insensitive. Do the examples of perfect identity always include mixed case sequence?

**Comment by lsterck** [comment]
> In this case, the overlap is only 11 nucleotides, so the merge is not made.

OK, makes sense. Will this affect the assembly end result? or will the worst effect be that sequence is "duplicated"?

**Comment by sjackman** [comment]
It shouldn't affect the assembly other than creating this 11 bp duplication. You should see in your assembly
```
CATAGATTTACAAAAATCCaacaactataanAACAACTATAACATAGATTTACAAAAATCC
```
Notice the lower case `n` separating the two.

> ok, for the first case, but in the second one fits both thresholds, no?

Yes, if the two sequences were swapped. Unfortunately the first sequence is the end of the first contig, and the second sequence is the start of the second contig, so it looks like the real overlap is ~107 bp, but ABySS hasn't grabbed enough sequence before doing the alignment. I think it grabs *k*-1 bp default.

**Comment by lsterck** [comment]
OK, got it.
what about the IUPAC code ones? If those also fall under one of the  above cases, you can safely close this issue from my part.

**Comment by lsterck** [comment]
perhaps a small improvement tip:

> In this case, the overlap is only 11 nucleotides, so the merge is not made.

then why not report so in the verbose output? something like 'overlap too small', rather then 'not matching'?

**Comment by lsterck** [comment]
should have know you had likely foreseen this somewhere in the code @sjackman :-) , but only if I request the extreme verbose output I see. Anyway, good enough.

**Comment by sjackman** [comment]
> Unfortunately the first sequence is the end of the first contig, and the second sequence is the start of the second contig, so it looks like the real overlap is ~107 bp, but ABySS hasn't grabbed enough sequence before doing the alignment. I think it grabs k-1 bp default.

@benvvalk Larger than *k*-1 bp overlaps aren't unusual when heterozygous sequence is involved. Perhaps `MergeContigs` should grab the last 2*k* or 3*k* of sequence to align.

**Comment by benvvalk** [comment]
@sjackman That makes sense.  I agree, it would be nice to improve the overlapping logic.

**Comment by mmokrejs** [comment]
```
warning: the head of 154467949- does not match the tail of the previous contig
CATAGATTTACAAAAATCCaacaactataa
                   |||||||||||
                   AACAACTATAACATAGATTTACAAAAATCC
```

@sjackman Would it be possible to change the wording of the sentence? I always looked for some problematic part of the sequence reported in reverse-complement of the latter and never realized what was the message about. By reading github issues I learned what you mean. It hurts to learn you swapped former and latter and moreover, you output both sequences in forward orientation. Please change the message the following:

```
warning: the tail of previous contig $contigname1 does not match head of 154467949- 
5'-CATAGATTTACAAAAATCCaacaactataa-3'
                   |||||||||||
                   5'-AACAACTATAACATAGATTTACAAAAATCC-3'
```

Thank you.

**Comment by mmokrejs** [comment]
> > Unfortunately the first sequence is the end of the first contig, and the second sequence is the start of the second contig, so it looks like the real overlap is ~107 bp, but ABySS hasn't grabbed enough sequence before doing the alignment. I think it grabs k-1 bp default.
> 
> @benvvalk Larger than _k_-1 bp overlaps aren't unusual when heterozygous sequence is involved. Perhaps `MergeContigs` should grab the last 2_k_ or 3_k_ of sequence to align.

This is embarrassing to me. So in my `k=160` assemblies I have not only `....${seq}n${seq}...` where `len($seq) < 20` duplications but even as long as `len($seq) < 159`? Is the distance information so inexact that abyss could not realize even an 11nt overlap is not sufficient? I mean, there must be plenty of Illumina paired-end pairs spanning say 200-500 range.

**Comment by sjackman** [comment]
@mmokrejs I think you misread the previous comments. In your case, the overlap is not merged because the minimum overlap is 20 nt, and your overlap is 11 nt.

https://github.com/bcgsc/abyss/blob/d4b4b5d3091d90a4967180d987bd7168dbf04585/MergePaths/MergeContigs.cpp#L96-L100
https://github.com/bcgsc/abyss/blob/d4b4b5d3091d90a4967180d987bd7168dbf04585/MergePaths/MergeContigs.cpp#L238-L239

If you use `abyss-pe v=-vvv` (or `MergeContigs -vvv`) you'll see a more helpful message. I agree though that the above message could be improved to say `insufficient overlap` rather than `does not match`.

**Comment by sjackman** [comment]
> So in my k=160 assemblies I have not only ....${seq}n${seq}... mis-duplications where len($seq) < 20 but even as long as len($seq) < 159?

Merges are made when len($seq) ≥ 20 and len($seq) < *k*. Overlaps *k* nt and larger are not typical. Because ABySS is a de Bruijn Graph assembler, overlaps are expected to be exactly *k*-1 nt. Other overlaps are seen only in unusual cases.

**Comment by lsterck** [comment]
I'll shamelessly revive a part of my original post:

> An additional remark is that I think this also is caused by the use of IUPAC code bases, as shown below:

```
warning: the head of 177027237+ does not match the tail of the previous contig
AAATTAGCTTGAGACCTGGTGGAGGAAGCAAATCGGGCTCTAACGAGAACCAAATTGGCTCGAGACCTGGTGGAGGAAGCAAAT
aacgagaaccAAATTRGCTYGAGACCTGGTGGAGGAAGCAAATCGGGCTCTAACGAGAAYCAAATTRGCTCGAGACCTGGTGGA
```

**Comment by sjackman** [comment]
Because it's an overlap alignment and the first and last ten bases are missing due to not taking enough of the sequence, the first and last are penalized gaps, so 74 / 94 = 78% identity, which is insufficient.
ABySS is aligning the last *k*-1 of the previous sequence to the first *k*-1 of the next sequence using an overlap alignment, where unaligned sequences at the beginning of the first sequence and at the end of the second sequence are not penalized. IUPAC ambiguity codes are handled correctly and are not penalized as mismatches when it an choose an appropriate base.

**Comment by sjackman** [comment]
If you're interested in writing code, here's the lines that would need to be patched.
https://github.com/bcgsc/abyss/blob/d4b4b5d3091d90a4967180d987bd7168dbf04585/MergePaths/MergeContigs.cpp#L226-L230
`ao` and `bo` (A overlap and B overlap) are the two sequences that are being aligned to each other.

**Comment by lsterck** [comment]
Would love the write code but unfortunately C is not in my programming repertoire :(

**Comment by lsterck** [comment]
I also still don't completely get the "issue" I think. 
I do get your explanation of why that last mentioned overlap is not accepted. However, I don't get what the role of the k-1 is in this? Do you mean that it could still be a 100% identical if for instance it took the last/first 100n but it now considers it as 'gap' because that bit of sequence has not been extracted from the original contigs? 
and how come then? because abyss does not expect the overlap to be bigger then k-1. is that it?

Out of curiosity: how would you 'solve' that? do progressive inward alignment?

**Comment by sjackman** [comment]
> Do you mean that it could still be a 100% identical if for instance it took the last/first 100n but it now considers it as 'gap' because that bit of sequence has not been extracted from the original contigs?

Yes.

> and how come then? because abyss does not expect the overlap to be bigger then k-1. is that it?

Yes.

> Out of curiosity: how would you 'solve' that? do progressive inward alignment?

We could align some arbitrary amount of sequence, like a kilobase, using our existing aligner.
We could align the two entire sequences to each other using a library like Heng Li's https://github.com/lh3/ksw2.

**Comment by mmokrejs** [comment]
I have also one example from abyss-2.1.5 here:

```
warning: the head of 5763445+ does not match the tail of the previous contig
CAAAACACCACCGGCCCACCGGGAAAAGTCCCGACTCTCCCGATGGCCACTCTGCCACTGCCTCCACTAGTCTGAAAGCAGAAGTGTTAAGCCCTGTATACTCGCCCGTCAGAAGAGGTGTTAAGCCCTGTGTACTCACCTGTCAGACGAAGTGTTAAGCCCTGTATACTCACCTGTCAGAAGAAGTGTTAAGCCTTGTATACTCACCTGTCAGATGAAGAGTTAAGCCCTGTATACTCACCCGTCAGAAGAGGTGTTAAGTCCTGTATACTCACCTGTCAGAAGAAGTGTTAAGCCCTGTATACTCATCTGTCAGAAGAAGTGTTAAGCCCTGTATACTCACCTGTCAGAAGAAGTGTTAAGCCCTATATACTCACCTGTCAGAAGAAGTGTTAAGCCCTGTATACTCACCTGTTAGAAGAGGTGTTAAGCCCTGTATGCTCACCTGTCAGAAGAAGTGTTAAGCCTTGTATACTCACCTGTCAGATGAAGAGTTAAGCCCTGTATACTTACCCGTCAGAAGAGGTGTTAAGTCCTGTAACTCACCTGTCAGAAGAAGTGTTAAGCCCTGTATACTCATCTGTCAGAAGAAGTGTTAAGCCCTTTATACTCACCTGTCAGAAGAAGTGTTAAACCCTGTATACTTGTCACCGGGTGAAAAGAGATGGGAACCTTTCTCACCCGGAACAAACAAACCGATCTAAAGTTTCCGCCGGACCATTCTTACCGAAACACCGGAAGTTCCGGTTCGCTCGTTCAAGCGAACATTAAAGGAAAATACGCAACAGGTGAGCTGGGAGATGTGGCGAGTGCGGATTGGCTGGCA
ggaacctttctcacccggaacaaacaaaccgatctaaagtttccgccggaccattcttaccgaaacaccggaagttccggttcgctcgttcaagcgaacattaaaggaaaatacgcaacaggtgagctgggagatgtggcgagtgcggattggctggcaCAAAACACCACCGGCCCACCGGGAAAAGTCCCGACTCTCCCGATGGCCACTCTGCCACTGCCTCCACTAGTCTGAAAGCAGAAGTGTTAAGCCCTGTATACTCGCCCGTCAGAAGAGGTGTTAAGCCCTGTGTACTCACCTGTCAGACGAAGTGTTAAGCCCTGTATACTCACCTGTCAGAAGAAGTGTTAAGCCTTGTATACTCACCTGTCAGATGAAGAGTTAAGCCCTGTATACTCACCCGTCAGAAGAGGTGTTAAGTCCTGTATACTCACCTGTCAGAAGAAGTGTTAAGCCCTGTATACTCATCTGTCAGAAGAAGTGTTAAGCCCTGTATACTCACCTGTCAGAAGAAGTGTTAAGCCCTATATACTCACCTGTCAGAAGAAGTGTTAAGCCCTGTATACTCACCTGTTAGAAGAGGTGTTAAGCCCTGTATGCTCACCTGTCAGAAGAAGTGTTAAGCCTTGTATACTCACCTGTCAGATGAAGAGTTAAGCCCTGTATACTTACCCGTCAGAAGAGGTGTTAAGTCCTGTAACTCACCTGTCAGAAGAAGTGTTAAGCCCTGTATACTCATCTGTCAGAAGAAGTGTTAAGCCCTTTATACTCACCTGTCAGAAGAAGTGTTAAACCCTGTATACTTGTCACCGGGTGAAAAGAGATG
2805724+ 842N 5656906+ 5763445+ 5647115-
The minimum coverage of single-end contigs is 0.92.
The minimum coverage of merged contigs is 3.82711.
Consider increasing the coverage threshold parameter, c, to 3.82711.
```

**Comment by sjackman** [comment]
The first sequence is a rotation of the second sequence. That is, one is AB, and the other is BA, and merging the two should produce ABA.
667 nucleotides match tail[0–667] and head[159–826]
159 nucleotides match tail[667–826] and head[0–159]
The first one is not a valid overlap alignment, but the second one is a valid overlap alignment of size 159 with 100% identity. I'm not sure why this merge was not made, but `MergeContigs -vvv …` should give you more information. It looks like a bug to me.

**Comment by mmokrejs** [comment]
Thank you, Shaun. Here are the details:

```
$ cat tt_16D1C3L12__abyss_160-6.fa tt_16D1C3L12__abyss_160-7.fa | MergeContigs -vvv -k160 -o tt_16D1C3L12__abyss_160-8.fa - tt_16D1C3L12__abyss_160-7.dot tt_16D1C3L12__abyss_160-7.path
Reading `tt_16D1C3L12__abyss_160-7.dot'...
Read 2925084 vertices. Using 356 MB of memory.
Reading `-'...
Read 1000000 sequences. Using 773 MB of memory.
Read 1462542 sequences. Using 2.06 GB of memory.
Reading `tt_16D1C3L12__abyss_160-7.path'...
Read 27470 paths. Using 2.07 GB of memory.

C---------------------------------------------------------------------------------------------------------------------------------------------------------------AAAACACCACCGGCCCACCGGGAAAAGTCCCGACTCTCCCGATGGCCACTCTGCCACTGCCTCCACTAGTCTGAAAGCAGAAGTGTTAAGCCCTGTATACTCGCCCGTCAGAAGAGGTGTTAAGCCCTGTGTACTCACCTGTCAGACGAAGTGTTAAGCCCTGTATACTCACCTGTCAGAAGAAGTGTTAAGCCTTGTATACTCACCTGTCAGATGAAGAGTTAAGCCCTGTATACTCACCCGTCAGAAGAGGTGTTAAGTCCTGTATACTCACCTGTCAGAAGAAGTGTTAAGCCCTGTATACTCATCTGTCAGAAGAAGTGTTAAGCCCTGTATACTCACCTGTCAGAAGAAGTGTTAAGCCCTATATACTCACCTGTCAGAAGAAGTGTTAAGCCCTGTATACTCACCTGTTAGAAGAGGTGTTAAGCCCTGTATGCTCACCTGTCAGAAGAAGTGTTAAGCCTTGTATACTCACCTGTCAGATGAAGAGTTAAGCCCTGTATACTTACCCGTCAGAAGAGGTGTTAAGTCCTGTAACTCACCTGTCAGAAGAAGTGTTAAGCCCTGTATACTCATCTGTCAGAAGAAGTGTTAAGCCCTTTATACTCACCTGTCAGAAGAAGTGTTAAACCCTGTATACTTGTCACCGGGTGAAAAGAGATGGGAACCTTTCTCACCCGGAACAAACAAACCGATCTAAAGTTTCCGCCGGACCATTCTTACCGAAACACCGGAAGTTCCGGTTCGCTCGTTCAAGCGAACATTAAAGGAAAATACGCAACAGGTGAGCTGGGAGATGTGGCGAGTGCGGATTGGCTGGCA
s..............................................................................................................................................................c....................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................gagatgggaacctttctcacccggaacaaacaaaccgatctaaagtttccgccggaccattcttaccgaaacaccggaagttccggttcgctcgttcaagcgaacattaaaggaaaatacgcaacaggtgagctgggagatgtggc..gtgcg..ttggc.g.ca
ggaacctttctcacccggaacaaacaaaccgatctaaagtttccgccggaccattcttaccgaaacaccggaagttccggttcgctcgttcaagcgaacattaaaggaaaatacgcaacaggtgagctgggagatgtggcgagtgcggattggctggcaCAAAACACCACCGGCCCACCGGGAAAAGTCCCGACTCTCCCGATGGCCACTCTGCCACTGCCTCCACTAGTCTGAAAGCAGAAGTGTTAAGCCCTGTATACTCGCCCGTCAGAAGAGGTGTTAAGCCCTGTGTACTCACCTGTCAGACGAAGTGTTAAGCCCTGTATACTCACCTGTCAGAAGAAGTGTTAAGCCTTGTATACTCACCTGTCAGATGAAGAGTTAAGCCCTGTATACTCACCCGTCAGAAGAGGTGTTAAGTCCTGTATACTCACCTGTCAGAAGAAGTGTTAAGCCCTGTATACTCATCTGTCAGAAGAAGTGTTAAGCCCTGTATACTCACCTGTCAGAAGAAGTGTTAAGCCCTATATACTCACCTGTCAGAAGAAGTGTTAAGCCCTGTATACTCACCTGTTAGAAGAGGTGTTAAGCCCTGTATGCTCACCTGTCAGAAGAAGTGTTAAGCCTTGTATACTCACCTGTCAGATGAAGAGTTAAGCCCTGTATACTTACCCGTCAGAAGAGGTGTTAAGTCCTGTAACTCACCTGTCAGAAGAAGTGTTAAGCCCTGTATACTCATCTGTCAGAAGAAGTGTTAAGCCCTTTATACTCACCTGTCAGAAGAAGTGTTAAACCCTGTATACTTGTCACCGGGTGAAAA--------------------------------------------------------------------------------------------------------------------------------------------------GA-----GA-----T-G--
666 / 985 = 0.676142 (too low)
warning: the head of 5763445+ does not match the tail of the previous contig
CAAAACACCACCGGCCCACCGGGAAAAGTCCCGACTCTCCCGATGGCCACTCTGCCACTGCCTCCACTAGTCTGAAAGCAGAAGTGTTAAGCCCTGTATACTCGCCCGTCAGAAGAGGTGTTAAGCCCTGTGTACTCACCTGTCAGACGAAGTGTTAAGCCCTGTATACTCACCTGTCAGAAGAAGTGTTAAGCCTTGTATACTCACCTGTCAGATGAAGAGTTAAGCCCTGTATACTCACCCGTCAGAAGAGGTGTTAAGTCCTGTATACTCACCTGTCAGAAGAAGTGTTAAGCCCTGTATACTCATCTGTCAGAAGAAGTGTTAAGCCCTGTATACTCACCTGTCAGAAGAAGTGTTAAGCCCTATATACTCACCTGTCAGAAGAAGTGTTAAGCCCTGTATACTCACCTGTTAGAAGAGGTGTTAAGCCCTGTATGCTCACCTGTCAGAAGAAGTGTTAAGCCTTGTATACTCACCTGTCAGATGAAGAGTTAAGCCCTGTATACTTACCCGTCAGAAGAGGTGTTAAGTCCTGTAACTCACCTGTCAGAAGAAGTGTTAAGCCCTGTATACTCATCTGTCAGAAGAAGTGTTAAGCCCTTTATACTCACCTGTCAGAAGAAGTGTTAAACCCTGTATACTTGTCACCGGGTGAAAAGAGATGGGAACCTTTCTCACCCGGAACAAACAAACCGATCTAAAGTTTCCGCCGGACCATTCTTACCGAAACACCGGAAGTTCCGGTTCGCTCGTTCAAGCGAACATTAAAGGAAAATACGCAACAGGTGAGCTGGGAGATGTGGCGAGTGCGGATTGGCTGGCA
ggaacctttctcacccggaacaaacaaaccgatctaaagtttccgccggaccattcttaccgaaacaccggaagttccggttcgctcgttcaagcgaacattaaaggaaaatacgcaacaggtgagctgggagatgtggcgagtgcggattggctggcaCAAAACACCACCGGCCCACCGGGAAAAGTCCCGACTCTCCCGATGGCCACTCTGCCACTGCCTCCACTAGTCTGAAAGCAGAAGTGTTAAGCCCTGTATACTCGCCCGTCAGAAGAGGTGTTAAGCCCTGTGTACTCACCTGTCAGACGAAGTGTTAAGCCCTGTATACTCACCTGTCAGAAGAAGTGTTAAGCCTTGTATACTCACCTGTCAGATGAAGAGTTAAGCCCTGTATACTCACCCGTCAGAAGAGGTGTTAAGTCCTGTATACTCACCTGTCAGAAGAAGTGTTAAGCCCTGTATACTCATCTGTCAGAAGAAGTGTTAAGCCCTGTATACTCACCTGTCAGAAGAAGTGTTAAGCCCTATATACTCACCTGTCAGAAGAAGTGTTAAGCCCTGTATACTCACCTGTTAGAAGAGGTGTTAAGCCCTGTATGCTCACCTGTCAGAAGAAGTGTTAAGCCTTGTATACTCACCTGTCAGATGAAGAGTTAAGCCCTGTATACTTACCCGTCAGAAGAGGTGTTAAGTCCTGTAACTCACCTGTCAGAAGAAGTGTTAAGCCCTGTATACTCATCTGTCAGAAGAAGTGTTAAGCCCTTTATACTCACCTGTCAGAAGAAGTGTTAAACCCTGTATACTTGTCACCGGGTGAAAAGAGATG
2805724+ 842N 5656906+ 5763445+ 5647115-

ATATCATTATTTACAGTTAAATAATATATGTTACACTGCAAACCTAAAATTTAAACCAAATCTAAAAATCACACCGATCACATGCGCTTGTGGGGGTACTTCAGACAGGATAAAGTGTTTTGGGGGTACAGTACCCTAAAAAGGTTGGGAACCACTGTTCTACGGTGTCCTCGCGCTCTGTCGAGCCTCCTCTAGAGACGATGTCCACCGTAGCATCGGAGGGAAGACTGTCAACCTCGGACACCGATCCAGACCCGCTCCCTCCTTCGGGTAGGTTTGCGGGGCCTGCGTTCGACCCCGAGATGGGAGCCATGCTCCCTCGGGCGGCGGACAGCTTGGAGTGGACTGA-ACCTCCCCCACCCGAACCATCGCGCCTCGATGATTGGTACTTCGGGGGTGCCAGGAGTTCTCAGTCCTCACCCC---CCCGTGCCCTTCTTCCCCGAGGTGCATGAAGAGTTGACACGGTCATGGAAAACC
                                                                                                                                                                                                                                                                                                                              ...................................................................................................................................................................
                                                                                                                                                                                                                                                                                                                              MTcgRKCggcggAYWRYTTRSAGTKRAMTRAtAYMTSYYMCACYSSAAccAtcgcgCCTcgAWRATTggTActtcgggggWRCCAggAgWTCTcARWMMTCACMCCgatCMCRTGCSCTTSTtcccSGRGGTRCWTSAagagttGACAcGGtcATggAAAaccSYGTTTTccgcccGGaaccGGccGTAtCAGccTtcAcCCCTMAcctctctMRAGGKTGGaGAtRCCAgSKGWTMTACtGGTRTCCtCGcgcTCWGTSGAGCSYCCKSTMGMGAYGcAgttgTGTCCgRCCgGTgcMGCWTCctcctGGWGGGAccAGcctACTSTCMMCtCWCGGRCWYSKAWSCAGWCttCYGSTttgaCgSCTSCTKCKKRYAGGgccTKTGSGGRGSCWGCgTTCgASCCCKRSATGSRAtGSCATKMYYSCTMRKRMRRCRKAtCARMTWRRAttGTGKWYTKAtAYYTMCYCttatagg
159 / 163 = 0.97546 (good)
The minimum coverage of single-end contigs is 0.92.
The minimum coverage of merged contigs is 3.82711.
Consider increasing the coverage threshold parameter, c, to 3.82711.
n	n:200	L50	min	N75	N50	N25	E-size	max	sum	name
1308579	964194	7975	200	2022	33563	109937	75097	908454	1.439e9	tt_16D1C3L12__abyss_160-8.fa
```

**Comment by sjackman** [comment]
The longer alignment has the larger alignment score.
-4*159 + 5*667 - 4*159 = 2063 (approximate alignment score, ignore gap open penalty)
5 * 159 = 795
The larger alignment has insufficient identity.
```
666 / 985 = 0.676142 (too low)
```
So it's not a bug, exactly, though undesirable behaviour perhaps.
